homeassistant/components/vicare/water_heater.py

```diff
@@ -0,0 +1,168 @@
+"""Viessmann ViCare water_heater device."""
+import logging
+
+import voluptuous as vol
+from PyViCare.PyViCareDevice import Device
+
+from homeassistant.components.water_heater import (
+    SUPPORT_TARGET_TEMPERATURE,
+    WaterHeaterDevice,
+)
+
+from homeassistant.const import (
+    TEMP_CELSIUS,
+    ATTR_TEMPERATURE,
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_NAME,
+    PRECISION_WHOLE,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ["PyViCare==0.1.0"]
+
+CONF_CIRCUIT = "circuit"
+
+VICARE_MODE_DHW = "dhw"
+VICARE_MODE_DHWANDHEATING = "dhwAndHeating"
+VICARE_MODE_FORCEDREDUCED = "forcedReduced"
+VICARE_MODE_FORCEDNORMAL = "forcedNormal"
+VICARE_MODE_OFF = "standby"
+
+VICARE_TEMP_WATER_MIN = 10
+VICARE_TEMP_WATER_MAX = 60
+
+OPERATION_MODE_ON = "on"
+OPERATION_MODE_OFF = "off"
+
+SUPPORT_FLAGS_HEATER = SUPPORT_TARGET_TEMPERATURE
+
+VICARE_TO_HA_HVAC_DHW = {
+    VICARE_MODE_DHW: OPERATION_MODE_ON,
+    VICARE_MODE_DHWANDHEATING: OPERATION_MODE_ON,
+    VICARE_MODE_FORCEDREDUCED: OPERATION_MODE_OFF,
+    VICARE_MODE_FORCEDNORMAL: OPERATION_MODE_ON,
+    VICARE_MODE_OFF: OPERATION_MODE_OFF,
+}
+
+HA_TO_VICARE_HVAC_DHW = {
+    OPERATION_MODE_OFF: VICARE_MODE_OFF,
+    OPERATION_MODE_ON: VICARE_MODE_DHW,
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
Please moved the configuration for both platforms to the component and make a `CONFIG_SCHEMA` there instead. Then load the platforms via `discovery.load_platform`.
https://github.com/home-assistant/home-assistant/blob/baa30aec9d903636c84b0acc41582e5197ee385a/homeassistant/helpers/discovery.py#L110